build-logic/src/main/kotlin/module.gradle.kts

```
@@ -57,12 +57,12 @@ tasks.withType<KotlinCompile>().configureEach {
             "-Xsuppress-version-warnings",
             "-opt-in=kotlin.RequiresOptIn"
         )
-        // Usage: <code>./gradlew build -PwarningsAsErrors=true</code>.
-        // Note: currently there are warnings for detekt-gradle-plugin that seemingly can't be fixed
+        // Note: Currently there are warnings for detekt-gradle-plugin that seemingly can't be fixed
         //       until Gradle releases an update (https://github.com/gradle/gradle/issues/16345)
+        // To ensure CI ~ Local build cache hit, we shouldn't use system env CI.
         allWarningsAsErrors = when (project.name) {
             "detekt-gradle-plugin" -> false
-            else -> (project.findProperty("warningsAsErrors") == "true" || System.getenv("CI") == "true")
+            else -> true
```

```suggestion
            else -> project.findProperty("warningsAsErrors") == "true"
```

I think that we shouldn't have this enable by default. It's a pain to develop with this enabled on local.